api/include/opentelemetry/context/context_value.h

```diff
@@ -13,7 +13,8 @@
 OPENTELEMETRY_BEGIN_NAMESPACE
 namespace context
 {
-using ContextValue = nostd::variant<bool,
+using ContextValue = nostd::variant<nostd::monostate,
```

Comment:
We don't currently run CI around `std::` substitute for `nostd::` . But I think this would have to be added around here: https://github.com/open-telemetry/opentelemetry-cpp/blob/d8467a78b3cf3accfa3a0a2569bfd3b90822ad02/api/include/opentelemetry/nostd/variant.h#L18

For the build flavor that uses either `absl::` or `std::` for `variant` implementation.

Reply:
If you can try building your change with `tools\build.cmd` on Windows just to check - it builds both `nostd` flavor and STL flavor (the one that aliases all `nostd::` classes to `std::` for vs2019 in C++20 mode).. Otherwise if all other folks are OK with the change, I can fix it if you are breaking `WITH_STL` option.

Reply:
+1 to support absl::monostate and std::monostate. 
@maxgolov  - What do you think about supporting only one implementation of variant (#572)

Reply:
Thanks @maxgolov and @lalitb , added alias for `nostd::monostate` to both `std::monostate` and `absl::monostate`.

Reply:
I'd say we switch to Abseil by default.